### DISCUSSION TITLE
Language choice about HoraeCTL

### DISCUSSION THREAD

**Discussion by jiacai2050** [CeresDB/ceresdb#1485] [Polls]
Hi community,

We plan to introduce a new component(called HoraeCTL) to help developers inspect status of HoraeDB server instance, it can do:
1. Check shard status
2. Check table status
3. Repair a broken table
4. ...

For now, we need to decide which language to use for this component, mainly we have two options:

# Go
- Simple to learn, easy to get started, this may be good for newcomers.

# Rust
- Unified project technology stack, HoraeDB is mainly a Rust project.

> The vote will end at 2024-02-25 UTC.

**Comment by tisonkun** [comment]
If we keep it in this repo, I'd prefer Rust. Rustup is a good project and I learn a lot form it to be confident that Rust can do this CTL job.

But if we create a new repo for it, Golang is a good choice also since [TiUP](https://github.com/pingcap/tiup) and [gtctl](https://github.com/GreptimeTeam/gtctl) show clear path to achieving it.

Generally, I vote for Rust and keep in the same repo.

**Reply by jiacai2050** [reply]
It's in the same repo.

**Comment by jiacai2050** [comment]
Others may concern that using rust will make it hard to integrate with k8s, in fact there is a k8s binding for rust, so it's not a real blocking issue to adopt rust.

- https://github.com/kube-rs/kube
> Hosted by [CNCF](https://cncf.io/) as a [Sandbox Project](https://www.cncf.io/sandbox-projects/)

**Comment by tisonkun** [comment]
So the conclusion is that we'd implement the CTL in this repo using Rust?

**Reply by jiacai2050** [reply]
Yes.
https://github.com/apache/incubator-horaedb/pull/1481
